Abstracting data pipline, CNN and RNN pre-processing and fixing syntaxis problems.

imgmaterial opened the pull request:
Majority of the work that was done is getting the code to run again and abstracting various parts of the data processing to allow writing separate scripts for data processing. I removed majority of hard coded paths from the code. A large part of the changes seen in the code is file renames, this was done to allow the files to be easily imported into new scripts.

imgmaterial committed:
Separated the data extractor into class difinition and a script executing the extraction processs

imgmaterial committed:
Worked through the TS scripts and turned them into functions that allow for file path input and output to be changed, wrote a script that uses the new functions

imgmaterial committed:
Fixed syntax errors in all of the RNN-PreProcessing scripts

imgmaterial committed:
Abstracted RNN pre-processing into functions and fixed syntax errors

imgmaterial committed:
file renames and rnn model files separated

imgmaterial committed:
Reorganized folders for CNN, abstracted the variouse CNN pre-processing scripts into functions, corrected some syntax errors and updated some the script to read the csv data

imgmaterial committed:
abstracting pre-processing for cnn and reorgonising folders so it would be easyer to import the new function

imgmaterial committed:
code for producing confusion confusion matrix

bryngemark commented on DataExtractor.py:
```diff
@@ -106,15 +77,15 @@ def Interloper(self):#Interloper is responsible for extracting the Energy and th
                  else:
-                          IDF.to_csv("/home/jacoblindahl/EcalID{}.csv".format(self.OutputNumber),mode='a',chunksize=5000,index=False, header=False)#This simply appends the data to the already created file (mode='a')
-                          EDF.to_csv("/home/jacoblindahl/Energy{}.csv".format(self.OutputNumber),mode='a',chunksize=5000,index=False, header=False)
+                          IDF.to_csv("Directory/EcalID{}.csv".format(self.OutputNumber),mode='a',chunksize=5000,index=False, header=False)#This simply appends the data to the already created file (mode='a')
```
How do you handle this when you're running the code, are you replacing `Directory` by something else? Consider making the path a variable too, that can be read from an input argument to the script.

bryngemark commented on ETSXRNN_ANN.py:
```diff
@@ -104,13 +104,13 @@ def __getitem__(self, idx) :#This function actually loads the data
     
     for file_name in batch_x:#Loads each file with the filenames included 
-        A = np.load('DataDirectory/' + str(file_name))
+        A = np.load('Output/Pre-Processing/RNN/ETSX/' + str(file_name))
```
We can think about paths two ways. Either, this is something the user always chooses for themself. Or, it's a larger structure within the code. In the latter case, there should be some setup steps or directory creation/existence checks inside the code being run. In the former cae, we should let all these be variables that can be set by reading an input variable passed to the script, or parsed from some general configuration file.

bryngemark commented on ETSXRNN_ANN.py:
```diff
@@ -153,16 +153,16 @@ def __getitem__(self, idx) :#This function actually loads the data
 
 
-lstm_2 = keras.layers.GRU(32, activation='tanh',return_sequences=True,recurrent_dropout=0.3)(lstm_1)#This layer also has recurrent dropout added in the layer 
+lstm_2 = keras.layers.GRU(32, activation='tanh',return_sequences=True,recurrent_dropout=0)(lstm_1)#This layer also has recurrent dropout added in the layer 
```
Not sure the comment about the dropout is true anymore?

bryngemark commented on ExtractDataScript.py:
```diff
@@ -0,0 +1,22 @@
+              "/projects/hep/fs9/shared/ldmx/ldcs/gridftp/mc23/v14/4.0GeV/v3.2.2-3e-v14/",
+              "/projects/hep/fs9/shared/ldmx/ldcs/gridftp/mc23/v14/4.0GeV/v3.2.2-4e-v14/"]
+event_count = [1,2,3,4]
+max_length = [155,250,360,450]
```
As we discussed, consider making them all fixed length

bryngemark replied:
dealt with later in the training code

bryngemark commented on AuroraJobEx.sh:
```diff
@@ -5,4 +5,4 @@
 #SBATCH --mem-per-cpu=6000 #This tells the system how much memory should be used per CPU involved in the calculation of the job 
 #SBATCH -N 2 #This specifies how many nodes should be used for the job 
```
Do you need two full nodes (40 cores) to be reserved for your job? I would have thought you could do single-core, or is this multithreaded in some way?

bryngemark replied:
I'll merge this in now, but this question still stands; just want to check that you aren't requesting more resources than you nee since that leads to unnecessary waiting time for both you and others

bryngemark reviewed (commented):
General comment: there is one python file per multiplicity. Looking superficially at it, they seem to run the same thing, with some array lengths differing, and likely some indices too? How difficult would it be to make this one single file, with those differences handled by input arguments to a function?

The same comment applies to the repeated loops in many of the files, these might be extracted into separate functions (could just be defs in the same file) and the multiplicity could be an input argument. I think this would make the code more readable and less error prone (avoids copy/paste errors)

imgmaterial committed:
added output path as a parameter to data extractor class

imgmaterial committed:
Condenced the ecal pre-processing for RNN and CNN into two small functions that take multiplicity as an input. Fixed a bug with the pre-processing where only 1 in a 1000 events would be saved to a file.

imgmaterial committed:
deleted the redundant pre-processing scripts for CNN and RNN ecal

imgmaterial commented:
Thank you for the review! I have addressed some but far from all of the problems in the last contributions. I have added the output path as a parameter to the DataExtractor class. The pre-processing for Ecal part of both the CNN and RNN has been condensed into two files. Along with that I tried to make it more readable and fixed a bug that resulted in not all events being saved after pre-processing. 
After diving a bit more into the code, the problem with differing length has already been addressed in the training scripts as RNN training pads the events with extra zeros before sending them to training.  
Most of the repeated loops are a part of processing scripts working with trigger scintillator information so I will condense those when the changes to ldmx-sw become available.

bryngemark reviewed (commented):
This looks a lot smarter now, with the condensed functions, and easier to work with.

bryngemark commented:
are you ready to have this merged in for now or are there more things you are still working on?

imgmaterial commented:
It can be merged in for now
